### PR TITLE
Use resolveRef to set refs in ScrollablePane

### LINKED ISSUE #3993: ScrollablePane using setTimeout instead of _async.setTimeout
### Bug Report

- __Package version(s)__: All versions
- __Browser and OS versions__: N/A

#### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? Yes

Requested priority: Normal

Products/sites affected: Anyone using the `ScrollablePane` component

#### Describe the issue:

When a `ScrollablePane` is unmounted before the timeout callback that is set in `componentDidMount` gets a chance to run, the callback will attempt to access `refs`, which are `null`, causing an exception to be thrown.
This does not break the app because the component has been unmounted already, but it does clutter the console and any exception handlers.

#### Actual behavior:

The global `setTimeout` is used, causing the timeout to run even if component has been unmounted already.

#### Expected behavior:

`_async.setTimeout` is used, ensuring the timeout is cleared when the component is unmounted.

### If applicable, please provide a codepen repro:

https://codepen.io/antonafMSFT/pen/VQrmOX?editors=1010
Repro:
Ensure console is open.
Click the Show/Hide button several times in a very quick succession.
Observer the null-reference exception.
The esception happens because the callback of `setTimeout` that is set in `componentDidMount` executes after the component has actually unmounted and the refs are all null.

### PR DESCRIPTION
**Pull request by antonafMSFT**
Use `resolveRef` to set refs in `ScrollablePane`

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/ScrollablePane/ScrollablePane.tsx [hunk 1]
```diff
@@ -83,13 +78,13 @@ export class ScrollablePane extends BaseComponent<IScrollablePaneProps, {}> impl
 
     return (
       <div
-        ref='root'
+        ref={ el => this._rootRef = el! }
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/ScrollablePane/ScrollablePane.tsx]: use resolveref:
```tsx
ref={ this._resolveRef('_rootRef') }
```

Comment (dzearing) [reply packages/office-ui-fabric-react/src/components/ScrollablePane/ScrollablePane.tsx]: And for the other refs.

Reasoning: This is the consistent pattern we have for resolving refs, which will create a function that's cached and won't regenerate on every re-render.

Also, React 16.3 will introduce a new more formal way of creating refs, and find/replace all the `_resolveRef` instances will be easier if they're consistent :)

File: packages/office-ui-fabric-react/src/components/ScrollablePane/ScrollablePane.tsx [hunk 2]
```diff
@@ -22,12 +22,10 @@ export class ScrollablePane extends BaseComponent<IScrollablePaneProps, {}> impl
     scrollablePane: PropTypes.object
   };
 
-  public refs: {
-    root: HTMLElement;
-    stickyContainer: HTMLElement;
-    stickyAbove: HTMLElement;
-    stickyBelow: HTMLElement;
-  };
+  private _rootRef: HTMLDivElement;
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/ScrollablePane/ScrollablePane.tsx]: Add change file.

**Comment by antonafMSFT** [comment]
Never mind. Another change has been approved that does this same thing while this one was waiting for approval.

**Pull request closed without merging**